bake/bake.go

```diff
@@ -233,7 +235,7 @@ func (c Config) target(name string, visited map[string]struct{}) (*Target, error
 		}
 	}
 	t.Inherits = nil
-	tt = merge(merge(defaultTarget(), t), tt)
+	tt = merge(merge(merge(defaultTarget(), tt), t), overrides[name])
```

Comment:
@tonistiigi I found another bug (not a regression from this PR). See test with VAR_BOTH. I believe this is correct: `t` is the child target, and `tt` is the parent, we want to merge the child into the parent, and the overrides on top.

Reply:
An important invariant I can add a comment in the code for, is that `t` should not have the default overrides (context = "." and dockerfile = "."), which is the case here because target() is called before settings those defaults in ResolveTarget.